lib/ArrayCache.php

```diff
@@ -85,7 +88,12 @@ public function set(string $key, string $value, int $ttl = null): Promise {
         } else {
             throw new \Error("Invalid cache TTL ({$ttl}; integer >= 0 or null required");
         }
-
+        if (isset($this->sharedState->cache[$key])) {
```

Comment:
The `isset` check can be avoided, `unset` doesn't complain if an item isn't set.